### DISCUSSION TITLE
No onboarding PR after 10+ hours

### DISCUSSION THREAD

**Discussion by JohNan** [renovatebot/renovate#17015] [Request Help]
Hi.

I have enabled the renovate bot on a second repository in my organization but have not got the onboarding PR. I tried removing/adding the repository again but no change. The repository shows up in the dashboard but there are no runs. As stated in the documentation I should request a manual start from staff.

**Comment by rarkins** [comment]
We can check it out but we'll need the name

**Reply by JohNan** [reply]
Thanks. It's a private repository: `Caspeco/BSTL-Core`

**Reply by rarkins** [reply]
The jobs are all timing out for this repo. I tried a dry run for this locally and see messages like this:

```
DEBUG: Dependency lookup unauthorized. Please add authentication with a hostRule (repository=Caspeco/BSTL-Core)
       "failedUrl": "https://REDACTED.pkgs.visualstudio.com/_packaging/Y
REDACTED/maven/v1/releases/org/slf4j/slf4j-simple/maven-metadata.xml"
```

So possibly all these failed lookups are causing the job to time out. Do you use this same registry for your other repo(s) successfully, e.g. by adding hostRules?

**Reply by rarkins** [reply]
When run with `--dry-run=lookup` it produces these stats:

```
       "hostStats": {
         "api.github.com": {
           "requestCount": 198,
           "requestAvgMs": 1846,
           "queueAvgMs": 0
         },
         "REDACTED.pkgs.visualstudio.com": {
           "requestCount": 90,
           "requestAvgMs": 137,
           "queueAvgMs": 0
         },
         "dl.google.com": {"requestCount": 843, "requestAvgMs": 102, "queueAvgMs": 0},
         "maven.pkg.jetbrains.space": {
           "requestCount": 651,
           "requestAvgMs": 280,
           "queueAvgMs": 0
         },
         "oss.sonatype.org": {
           "requestCount": 5457,
           "requestAvgMs": 327,
           "queueAvgMs": 0
         },
         "plugins.gradle.org": {
           "requestCount": 2673,
           "requestAvgMs": 191,
           "queueAvgMs": 0
         },
         "repo.maven.apache.org": {
           "requestCount": 2131,
           "requestAvgMs": 60,
           "queueAvgMs": 0
         },
         "rubygems.org": {"requestCount": 1, "requestAvgMs": 3950, "queueAvgMs": 0},
         "s01.oss.sonatype.org": {
           "requestCount": 128,
           "requestAvgMs": 327,
           "queueAvgMs": 0
         },
         "services.gradle.org": {
           "requestCount": 1,
           "requestAvgMs": 159,
           "queueAvgMs": 0
         }
       },
       "totalRequests": 12173
 INFO: Repository finished (repository=Caspeco/BSTL-Core)
       "durationMs": 1582691
```

**Reply by JohNan** [reply]
Thank you. I removed that particular private maven repo. Please try again.

I do have the same registry in other repos and I have added special rules for it. But I didn't think that the bot would do the onboarding PR if I added a `renovate.json` file to the repo. I can add it if you think that would help?

edit: `renovate.json` file added with `hostRules`, just in case.

**Reply by rarkins** [reply]
Yes onboarding should be skipped if you add renovate.json to the default branch

**Reply by JohNan** [reply]
Ok. So what do you suggest? I still don't see any runs in the dashboard.

**Reply by rarkins** [reply]
I'll check again tomorrow

**Reply by rarkins** [reply]
Still timing out each run unfortunately

**Reply by JohNan** [reply]
How strange. Did you get the same failed lookups this time also?

**Reply by rarkins** [reply]
Currently when an app times out like this we are unfortunately not getting the logs for it. It's something we're looking into now whether we can "fix" (i.e. to capture at least the logs which occurred).

When I run it locally, I don't have a copy of the app's private key in order to decrypt encrypted hostRules values, so I still see the failures. I decided to abort the local test once I saw that happening. It's my choice to never copy the app's private key out of its secure vault so therefore this is not something I can fully reproduce.

**Reply by rarkins** [reply]
I might try setting the repo to "dry run" to see if that can complete successfully

**Reply by JohNan** [reply]
Ok. Is there anything I can do to help? I'm changing the secret to the exact same as I have in another repo, just to be sure it's working.

**Reply by JohNan** [reply]
I just realized that the url I added in the hostRules was not correct. I have corrected it by adding another rule.

**Reply by rarkins** [reply]
Now there's an error: `Encrypted secret is scoped to a different repository: "Caspeco/BSTL-REDACTED".`

You'll need to re-encrypt it with this new repo name, and update the `renovate.json`

**Reply by rarkins** [reply]
FYI it's running in dry run mode currently so likely didn't make an issue in your repo about this, but you can hopefully see a log file now through the dashboard

**Reply by JohNan** [reply]
Ok. Try now again. If still can't decrypt it will have to wait until Monday when I'm back at work.

Thanks for the help so far!

**Reply by rarkins** [reply]
It should try itself after a few hours. I'm afk for a few hours too so can't trigger it manually

**Reply by rarkins** [reply]
It seems we can now capture logs of timed out jobs in such situations so I have removed the dryRun settings for your repo in order to try again. It may still time out, but at least we should be able to see where

**Reply by JohNan** [reply]
Great news! Are you able to trigger another run manually?

**Reply by JohNan** [reply]
I can now see the runs but the logs are not appearing and the status is still `timeout`. The logs are loading forever and the request shows `{message: "log file not found"}` in the network tab.

I tried running the renovate docker container locally and it hangs just after fetching change logs. When comparing to runs from a different repo it is just before this step

```
DEBUG: Fetching changelog: https://github.com/JetBrains/kotlin (1.7.10 -> 1.8.0-dev-1486)
-- HANGS HERE --
DEBUG: config.repoIsOnboarded=true
DEBUG: packageFiles with updates
```

I have also disable the private registry host in `hostRules`. It shouldn't even make lookups to that registry because it is set as a publishing registry.

```
publishing {
    repositories {
        maven {
            url = if (version.toString().endsWith("-SNAPSHOT")) {
                uri("https://REDACTED.pkgs.visualstudio.com/REDACTED/snapshots")
            } else {
                uri("https://REDACTED.pkgs.visualstudio.com/REDACTED/releases")
            }
            credentials {
                username = "REDACTED"
                password = REDACTED
            }
        }
    }
}
```

**Reply by rarkins** [reply]
Could you file a bug report issue with reproduction for the publishing registry problem?

**Reply by rarkins** [reply]
I've also noted that changelogs appear to be the problem. For repos which have been timing out, I am now trying disabling release notes using server side config (`fetchReleaseNotes`: false).

**Reply by JohNan** [reply]
Im not sure if it was something that you did or the fact that I updated a bunch of dependencies manually. When I ran it locally it found `28 flattened updates` so I figured that maybe it was too many. I fixes a bunch manually and was down to `13 flattened updates` and it started to work.

**Reply by rarkins** [reply]
It was probably the changelog suppression

**Reply by JohNan** [reply]
Ok. Thanks for sorting it out. We can live without the changelog for now.

I filed a bug report regarding the publishing registries: https://github.com/renovatebot/renovate/issues/17067

**Reply by rarkins** [reply]
We are working on refactoring changelogs and hopefully at some point we'll discover a performance optimization that lets us reenable them for your repo again
